core/domain/opportunity_services.py

```diff
@@ -0,0 +1,370 @@
+            opportunity_domain.ExplorationOpportunitySummary(
+                exp_id, topic.id, topic.name, story.id, story.title, node.title,
+                content_count, copy.deepcopy(incomplete_translation_languages),
+                translation_counts,
+                copy.deepcopy(need_voiceartist_in_languages), []))
+
+        exploration_opportunity_summary_list.append(
+            exploration_opportunity_summary)
+
+    _save_multi_exploration_opportunity_summary(
+        exploration_opportunity_summary_list)
+
+
+def update_exploration_opportunity_with_new_exploration(
+        exp_id, content_count, translation_counts,
+        complete_translation_language_list):
+    """Updates the opportunities models with the changes made in th exploration.
+
+    Args:
+        exp_id: str. The exploration id which is also the id of the opportunity
+            model.
+        content_count: int. The number of content available in the exploration.
+        translation_counts: dict. A dict with language code as a key and
+                number of translation available in that language as a value.
+        complete_translation_language_list: list(str). A list of language code
+            in which the exploration translation is 100%.
+    """
+    model = opportunity_models.ExplorationOpportunitySummaryModel.get(exp_id)
+    exploration_opportunity_summary = (
+        get_exploration_opportunity_summary_from_model(model))
+    exploration_opportunity_summary.content_count = content_count
+    exploration_opportunity_summary.translation_counts = translation_counts
+    exploration_opportunity_summary.complete_translation_lanuages = (
+        complete_translation_language_list)
+
+    new_languages_for_voiceover = set(complete_translation_language_list) - set(
+        exploration_opportunity_summary.assigned_voiceartist_in_languages)
+
+    # We only append new languages to need_voiceartist_in_languages, as the
+    # complete translation languages list will be dynamic based on some
+    # content text are changed, where as the voiceover is a long term work and
+    # we can allow a voiceartist to work for an exploration which needs a little
+    # bit update in text translation.
+    need_voiceartist_in_languages_set = set(
+        exploration_opportunity_summary.need_voiceartist_in_languages)
+    need_voiceartist_in_languages_set |= set(new_languages_for_voiceover)
+
+    exploration_opportunity_summary.need_voiceartist_in_languages = list(
+        need_voiceartist_in_languages_set)
+
+    exploration_opportunity_summary.validate()
+
+    _save_multi_exploration_opportunity_summary(
+        [exploration_opportunity_summary])
+
+
+def update_exploration_opportunities_with_new_story(story, exp_ids):
+    """Updates the opportunities models with the new story.
+
+    Args:
+        story: Story. The new story object.
+        exp_ids: list(str). A list of exploration ids. Corresponding to these
```

Comment:
... of exploration IDs whose exploration opportunity summary models need to be updated.